src/hardware/adlib.cpp

```diff
@@ -889,28 +950,37 @@ Module::Module(Section *configuration)
 	const auto read_from = std::bind(&Module::PortRead, this, _1, _2);
 	const auto write_to = std::bind(&Module::PortWrite, this, _1, _2, _3);
 
-	// 0x388 range
+	// 0x388-0x38b ports (read/write)
 	constexpr io_port_t port_0x388 = 0x388;
 	WriteHandler[0].Install(port_0x388, write_to, io_width_t::byte, 4);
 	ReadHandler[0].Install(port_0x388, read_from, io_width_t::byte, 4);
-	// 0x220 range
+
+	// 0x220-0x223 ports (read/write)
 	if (!single) {
 		WriteHandler[1].Install(base, write_to, io_width_t::byte, 4);
 		ReadHandler[1].Install(base, read_from, io_width_t::byte, 4);
 	}
-	//0x228 range
+	// 0x228-0x229 ports (write)
 	WriteHandler[2].Install(base + 8u, write_to, io_width_t::byte, 2);
+
+	// 0x228 port (read)
 	ReadHandler[2].Install(base + 8u, read_from, io_width_t::byte, 1);
 
 	MAPPER_AddHandler(OPL_SaveRawEvent, SDL_SCANCODE_UNKNOWN, 0,
 	                  "caprawopl", "Rec. OPL");
 }
 
-Module::~Module() {
+Module::~Module()
+{
 	delete capture;
 	capture = nullptr;
 	delete handler;
 	handler = nullptr;
+
+	if (adlib_gold) {
+		delete adlib_gold;
+		adlib_gold = nullptr;
+	}
```

Comment:
(minor)

It's OK to `delete` a `nullptr`; but I  this is more clear/aware.

https://en.cppreference.com/w/cpp/language/delete

> "the default deallocation functions are guaranteed to do nothing when passed a null pointer. "

Reply:
Ah, good to know. I guess I'm still in the C mindset -- always do a null check before a `free` call, I learned that the hard way 😅

Reply:
I've made the change anyway.